### PR TITLE
Skip image build & push on PRs

### PR DESCRIPTION
**Pull request by vharsh**
## TODOs
- I'll later refactor CircleCI to use container executors & convenience images for golang which might help the builds end a bit faster.

(Q) Is it okay if we skip the `go build`?
(A) Yes, because `go test -cover ./...` would have already built it

```
The idea is simple: Rewrite the package's source code before compilation to add instrumentation, compile and run the modified source, and dump the statistics.
```
Read more: https://blog.golang.org/cover

Signed-off-by: Harsh Vardhan <harsh.vardhan@mayadata.io>

### CODE REVIEW

**Review by Sanjay1611 — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -58,6 +58,9 @@ jobs:
           command: bash <(curl -s https://codecov.io/bash)
           # Ref: https://circleci.com/docs/2.0/configuration-reference/#the-when-attribute
           when: always
+      - run:
```
Comment (avishnu) [inline .circleci/config.yml:61]: How does this work? I see a "run" block already below. Can we have 2 blocks with the same tag?

Comment (vharsh) [reply .circleci/config.yml:61]: Yes, circleCI can accept a list of steps(or maybe runs) on a workflow

**Review by avishnu — APPROVED** [review]
lgtm

File: .circleci/config.yml (hunk 1 above)
Comment (avishnu) [reply .circleci/config.yml:61]: ok

**Pull request merged**